### ISSUE TITLE
Increasing batch size

### ISSUE THREAD

**Issue by jaideep11061982** [zhiyuanyou/SAFECount#5]
@zhiyuanyou  is it possible to convert this into batch operation currently its a single image based both in training and inference.
1)can this converted to batch based operation.
2) Would increasing the batch size improve the gradients ?

3) I do not have DDP , i.e one gpu  only
can i comment this out in training

```
if distributed:
        
        dist.all_reduce(train_loss)
        dist.all_reduce(iter)
```

4) Now I am using backend as resnet50 and doing retraining  using its pretrained wts but I get too less training loss ,is it expected
```
Train | Epoch : 1 / 200 | Iter: 1 / 3659 | lr: 2e-05 | Data: 0.38, Time: 12.12 | Loss: 0.0008869824814610183
 Train | GT: 200.0, Pred: 299.8 | Best Val MAE: inf, Best Val RMSE: inf
 Train | Epoch : 1 / 200 | Iter: 801 / 3659 | lr: 2e-05 | Data: 0.00, Time: 0.18 | Loss: 0.2721385396662299
```

5) Did you train for 200 epochs ?

**Comment by zhiyuanyou** [comment]
Hi～
1. Actually not. Some operations only support _batch_size == 1_. 
2. The same as 1.
3. Yes. You could comment these lines. 
4. The scale of loss is reasonable. Finally, it could reach _1e-6_ even _1e-8_. 
5. 200 epochs for FSC-147.

**Comment by jaideep11061982** [comment]
@zhiyuanyou  
1) if I use a different image size say rectangular 512,768 . Which parameter do I have to adjust
as I get 0 for one of dimenstion of feat after using crop roi  function
2) How do you chose the outstride, do I have to change it depending on the image resize shape I chose ?

3) I put some debug statements before crop roi , I get this for 512,768
 ```
feat torch.Size([1, 256, 128, 192]) tensor([[216., 497., 241., 527.],
        [186., 403., 201., 425.],
        [330., 498., 366., 518.],
        [256., 245., 278., 279.]],
```
How does crop roi able to extract desired regions, as it positions overshot the feat shape on h,w dimensions

**Comment by zhiyuanyou** [comment]
Hi~ 
1. 0 dimension in _feature_size_ is because some boxes are too small. What dataset did you use? FSC-147 or your own dataset?
2. Choose _out_stride_ according to _feature_size = image_size / out_stride_.
3. No.
4. I do not understand. Could you please describe this issue clearer?
5. The boxes is in original _image_size_, we should convert it to _feature_size_, thus we need to divide _out_stride_.

**Comment by jaideep11061982** [comment]
@zhiyuanyou i figured out the logic. your crop roi and bbx transformation are based on the assumption that final fmap size will be /4   if there are changes into that then it disturbs BBx based crop computation in  ROI

**Comment by zhiyuanyou** [comment]
Yes. Cropping ROI is operated on feature map (whose size is _image_size_ / _out_stride_), thus we need to divide _out_stride_ first.
